api/provider/aws/service_read.go

```diff
@@ -55,12 +74,91 @@ func (s *ServiceProvider) readService(clusterName, serviceID string) (*ecs.Servi
 
 	output, err := s.AWS.ECS.DescribeServices(input)
 	if err != nil {
+		if err, ok := err.(awserr.Error); ok && err.Code() == "ServiceNotFoundException" {
+			return nil, errors.Newf(errors.ServiceDoesNotExist, "Service '%s' does not exist", serviceID)
+		}
+
+		return nil, err
+	}
+
+	if len(output.Services) == 0 {
+		return nil, errors.Newf(errors.ServiceDoesNotExist, "Service '%s' does not exist", serviceID)
+	}
+
+	return output.Services[0], nil
+}
+
+func (s *ServiceProvider) lookupDeployIDFromTaskDefinitionARN(taskDefinitionARN string) (string, error) {
```

Comment:
This should be in `common.go` since `task.Read()` will be using this.

Reply:
✅